content/client-lib-development-guide/features.textile

```diff
@@ -73,9 +73,10 @@ h3(#restclient). RestClient
-** @(RSC15b)@ This fallback behaviour only applies to the default @rest.ably.io@ endpoint; if either @host@ or @environment@ options are set, no fallback host is used
-** @(RSC15d)@ Automatic retries to fallback hosts will not occur for any HTTP status codes not in the range 500 to 504. For example, if the credentials are invalid (a 401 status code) or the maximum message length has been exceeded, then resending the request to an alternative data center will not fix the problem and will only increase the load on other data centers. See the "complete list of Ably error codes":https://github.com/ably/ably-common/blob/master/protocol/errors.json#L11 and a "discussion in the Java library about this":https://github.com/ably/ably-java/issues/14
+** @(RSC15a)@ All HTTP requests sent to Ably are fault-tolerant and will fallback to an alternative host name if the default host endpoint is unreachable or indicates that it is unserviceable with a 500 to 504 status code. This ensures that a client library is able to work around routing or other problems with the user's closest data centre. For example, if a @POST@ request to @rest.ably.io@ fails because the default endpoint is unreachable or unserviceable, then the @POST@ request should be retried again against the fallback hosts in attempt to find an alternate healthy data centre to service the request. The fallback hosts are @[a-e].ably-realtime.com@ and should be attempted in random order, with a maximum number of fallback attempts being the configured @httpMaxRetryCount@ (see "@TO3l5@":#TO3l5)
+** @(RSC15b)@ This fallback behaviour only applies to the default @rest.ably.io@ endpoint; if either @host@ or @environment@ options are set resulting in the endpoint not being @rest.ably.io@, no fallback host is used
+** @(RSC15d)@ Automatic retries to fallback hosts will not occur for any HTTP status codes not in the range 500 to 504. For example, if the credentials are invalid (a 401 status code) or the maximum message length has been exceeded, then resending the request to an alternative data centre will not fix the problem and will only increase the load on other data centres. See the "complete list of Ably error codes":https://github.com/ably/ably-common/blob/master/protocol/errors.json#L11
+** @(RSC15e)@ Every new HTTP operation is always first attempted to the primary host, and then, only upon "an appropriate request failure":#RSC15a, attempted against the fallback hosts when "RSC15b":#RSC15b applies. This ensures that should the closest data centre become available at a later time, the client library will automatically resume using the primary data centre at @rest.ably.io@
```

Comment:
I have no problem re-ordering, yes it's a bit more of a hassle but it's not a big problem...  I will have a go at that and addressing your points